packages/web-client/app/models/workflow/milestone.ts

```diff
@@ -9,15 +9,19 @@ interface MilestoneOptions {
 export class Milestone {
   title: string;
   postables: WorkflowPostable[] = [];
-  includedPostables = new Set<WorkflowPostable>();
-  excludedPostables = new Set<WorkflowPostable>();
+  #includedPostables = new Set<WorkflowPostable>();
+  #excludedPostables = new Set<WorkflowPostable>();
   workflow: Workflow | undefined;
   setWorkflow(wf: Workflow) {
     this.workflow = wf;
     this.postables.invoke('setWorkflow', wf);
   }
   get isComplete() {
```

Comment:
Would changing this to `this.visiblePostables.isEvery('isComplete', true)` make sense? To make the `isComplete` property less dependent on `visiblePostables` having recently been accessed.

Reply:
Hmm, not sure. visiblePostables has side effects -- see my comment here: https://github.com/cardstack/cardstack/pull/1564/files#diff-217fd8162cfba6de1caa5800dc8cd7a75e9ff7df25dffad73d3c55ae72dde876R30-R34

Reply:
Hmm, looks a bit tricky. If we don't change to `this.visiblePostables.isEvery('isComplete', true)`, then we can't reliably check milestone completion status before `Milestone.visiblePostables` is accessed right? I guess this would be relevant to something like this where we have a quick view of different workflows' completion statuses without rendering the whole thread:

![image](https://user-images.githubusercontent.com/39579264/114866348-04b74e80-9e26-11eb-8a10-943d694db38c.png)

But if we do, then we might add an incorrect timestamp to the postables for that milestone - or running code that adds the postables to `Milestone.#excludedPostables` because it's called before the milestone is 'ready'.

Maybe this is something we can wait to change, but make a comment on?

Reply:
Sorry, github looked like it doubled up my comment.

Can we leave a comment about not accessing `Milestone.isComplete` without accesssing `Milestone.visiblePostables`? It might be relevant for things like overviews of workflows:

![image](https://user-images.githubusercontent.com/39579264/114867228-19e0ad00-9e27-11eb-8feb-6591cc19609e.png)

Reply:
This is a really good point. I guess since the `visiblePostables` getter is idempotent, it would be OK to use from `isComplete` as well as the template. The only real issue maybe is the timestamp side effect, but maybe even that is OK.

Reply:
This will all change once we have workflow persistence, too.